Cython/Compiler/ModuleNode.py

```diff
@@ -229,7 +234,10 @@ def generate_public_declaration(self, entry, h_code, i_code):
                 entry.type.declaration_code(entry.cname, pyrex=1)))
 
     def api_name(self, env):
-        return env.qualified_name.replace(".", "__")
+        api_name = self.punycode_module_name(env.qualified_name)
+        if api_name.startswith('_'):
+            api_name = api_name[1:]
```

Comment:
It's now changed to be slightly different. I was trying not to change the api names generated for existing non-unicode modules. The version I've changed it to generated `import_name` for non-unicode names (as before) and `importU_punycodename` for unicode modules, which I think is reasonable. It's essentially because punycode_module_name returns something starting with "_"